src/tests/system/tests/test_ldap.py

```diff
@@ -21,9 +21,13 @@
 @pytest.mark.parametrize("use_ppolicy", ["true", "false"])
 @pytest.mark.topology(KnownTopology.LDAP)
 @pytest.mark.parametrize("sssd_service_user", ("root", "sssd"))
-@pytest.mark.require(
-    lambda client, sssd_service_user: ((sssd_service_user == "root") or client.features["non-privileged"]),
-    "SSSD was built without support for running under non-root",
+@pytest.mark.skipif(
+    bool(lambda client, sssd_service_user: ((sssd_service_user != "root") and not client.features["non-privileged"])),
+    reason="SSSD was built without support for running under non-root",
+)
+@pytest.mark.skipif(
+    bool(lambda client: not client.features["ldap_use_ppolicy"]),
+    reason="SSSD is missing support for ldap_use_ppolicy.",
 )
```

Comment:
Can we use `@pytest.mark.builtwith("ldap_use_ppolicy") instead?

Reply:
The problem with builtwith and require is that the skip is done in the fixture outside of the test context producing output like this:
```
SKIPPED [8] ../../../../../../../../.local/lib/python3.11/site-packages/pytest_mh/_private/fixtures.py:195: SSSD was built without support for running under non-root or ldap_use_ppolicy support is missing.
```
If I use the skipif decorator I get more usable output with testsuite and function line reference:
```
SKIPPED [8] ../sssd/src/tests/system/tests/test_ldap.py:17: ppolicy not supported.
```
The most useful output is when the skip is done in the test function body when the suite and test function is reported.

Reply:
Ok, approving this one for now.   We'll address the builtwith reporting separately and can address updates to use that later when some CI updates have been made.